crates/rspack_core/src/chunk_graph/chunk_graph_module.rs

```diff
@@ -163,4 +168,147 @@ impl ChunkGraph {
   ) {
     self.block_to_chunk_group_ukey.insert(block, chunk_group);
   }
+
+  pub fn get_module_graph_hash(
+    &self,
+    hash_options: &OutputOptions,
+    module: &BoxModule,
+    module_graph: &ModuleGraph,
+    with_connections: bool,
+  ) -> String {
+    let mut hasher = RspackHash::new(&hash_options.hash_function);
+    let mut connection_hash_cache: HashMap<Identifier, String> = HashMap::new();
+
+    fn update_export_info_hash(
+      hasher: &mut RspackHash,
+      export_info_id: &ExportInfoId,
+      module_graph: &ModuleGraph,
+    ) {
+      if let Some(export_info) = module_graph.export_info_map.get(export_info_id) {
+        export_info.name.dyn_hash(hasher);
+        export_info.module_identifier.dyn_hash(hasher);
+        export_info.usage_state.dyn_hash(hasher);
+        for (name, value) in &export_info.target {
+          name.dyn_hash(hasher);
+          value.dyn_hash(hasher);
+        }
+        for (name, value) in &export_info.max_target {
+          name.dyn_hash(hasher);
+          value.dyn_hash(hasher);
+        }
+        export_info.provided.dyn_hash(hasher);
+        export_info.can_mangle_provide.dyn_hash(hasher);
+        export_info.terminal_binding.dyn_hash(hasher);
+        export_info.target_is_set.dyn_hash(hasher);
+        export_info.max_target_is_set.dyn_hash(hasher);
+        export_info.exports_info_owned.dyn_hash(hasher);
+        if let Some(exports_info_id) = export_info.exports_info {
+          update_exports_info_hash(hasher, &exports_info_id, module_graph);
+        }
+      }
+    }
+
+    fn update_exports_info_hash(
```

Comment:
Can you move `exports_info` and `export_info` hash logic to `Hash` trait by self struct? It's is better if add some comment if some fields shouldn't hash, if somebody add a filed, it is easy to avoid forget hash it.